Fix sed script and add echo with newline

ahanafy opened the pull request:
Removed apostrophes from sed line, and added a newline when appending new team directory to kustomize.yaml file

ahanafy committed:
Fix sed script and add echo with newline

Removed apostrophes from sed line, and added a newline when appending new team directory to kustomize.yaml file

stefanprodan commented on scripts/create-team.sh:
```diff
@@ -15,9 +15,9 @@ cp -r "${REPO_ROOT}/cluster/${TEMPLATE}/." ${TEAM_DIR}
 for f in "${TEAM_DIR}*.yaml"
 do
- sed -i '' "s/$TEMPLATE/$TEAM_NAME/g" ${f}
+ sed -i "s/$TEMPLATE/$TEAM_NAME/g" ${f}
```
Have you tested this on macOS?

ahanafy replied:
I didn't test on OSX but i noticed there were some issues with sed compatibility.
I submitted a new change with a version of the sed command that is more compatible with OSX 10.9+ and GNU.
I don't have a mac, but if you do, could you confirm that the sed command works?

stefanprodan replied:
It doesn't work:
```
tree cluster/team2
cluster/team2
├── flux-patch.yaml
├── flux-patch.yaml-e
├── kubeconfig.yaml
├── kubeconfig.yaml-e
├── kustomization.yaml
├── kustomization.yaml-e
├── namespace.yaml
├── namespace.yaml-e
├── psp.yaml
├── psp.yaml-e
├── rbac.yaml
└── rbac.yaml-e
```

stefanprodan replied:
You could switch the commands based on `uname` and use the old command on Darwin.

ahanafy replied:
Did you test on 10.6 or 10.9+? The `uname` trick is what was listed in the stackoverflow thread, but what is the difference in `uname` output between 10.6 vs 10.9+? Is there some way to determine if 10.6 vs 10.9? I figured at least supporting 10.9+ was a good start. I think 10.6 is around 10 years old?

ahanafy committed:
Update sed command to be compatible with newer OSX

sed command worked on linux, but OSX needs different format for sed.
Unless more logic added this will work on OSX 10.9+ and GNU Linux.
https://stackoverflow.com/questions/4247068/sed-command-with-i-option-failing-on-mac-but-works-on-linux

ahanafy reviewed (commented):
This is the version that should work on 10.9+ and GNU

stefanprodan commented:
Closing in favour of #6
